### PR TITLE
Relax query builder validation for custom indexes

### PR DESCRIPTION
**Pull request by olim7t**
Simplest possible fix to allow SAI queries.

### CODE REVIEW

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- persistence-api/src/main/java/io/stargate/db/schema/SecondaryIndex.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6372866&bid=20678589)

File: persistence-common/src/main/java/io/stargate/db/datastore/common/AbstractCassandraSchemaConverter.java [hunk 1]
```diff
@@ -88,6 +88,9 @@
   /** The target of the provided internal index, that is the content of it's "target" option. */
   protected abstract String indexTarget(I index);
 
+  /** Whether the given index has is of the "CUSTOM" kind. */
```
Comment (EricBorczuk) [inline persistence-common/src/main/java/io/stargate/db/datastore/common/AbstractCassandraSchemaConverter.java:91]: super nit: `given index is of the`

**Review by EricBorczuk — APPROVED** [review]
Seems good to me for the time being!

File: persistence-api/src/main/java/io/stargate/db/schema/SchemaBuilderImpl.java [hunk 2]
```diff
@@ -338,6 +339,11 @@ public void indexFull() {
     indexFull = true;
   }
 
+  @DSLAction
+  public void indexCustom() {
```
Comment (dimas-b) [inline persistence-api/src/main/java/io/stargate/db/schema/SchemaBuilderImpl.java:343]: Should it also be added to the DSL grammar (the `@GenerateEmbeddedDSL` annotation on this class)?

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- persistence-api/src/main/java/io/stargate/db/schema/SecondaryIndex.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6372866&bid=20678589)

**Review by dimas-b — APPROVED** [review]

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- persistence-api/src/main/java/io/stargate/db/schema/SecondaryIndex.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6372866&bid=20678589)

**Review by dougwettlaufer — APPROVED** [review]

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- persistence-api/src/main/java/io/stargate/db/schema/SecondaryIndex.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6372866&bid=20678589)

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- persistence-api/src/main/java/io/stargate/db/schema/SecondaryIndex.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6372866&bid=20678589)

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- persistence-api/src/main/java/io/stargate/db/schema/SecondaryIndex.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6372866&bid=20678589)

**Review by jorgebay — APPROVED** [review]

**Review by mpenick — APPROVED** [review]

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- persistence-api/src/main/java/io/stargate/db/schema/SecondaryIndex.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6372866&bid=20678589)

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- persistence-api/src/main/java/io/stargate/db/schema/SecondaryIndex.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6372866&bid=20678589)

**Pull request merged**